_plugins/idp_translations.rb

```diff
@@ -0,0 +1,34 @@
+## This plugin copies the fonts and the images from the identity style guide
+## to the appropriate path on the built site
+##
+## The CSS and JS is already handled by Jekyll and the build:js npm script
+
+require 'fileutils'
+require 'uri'
+require 'net/http'
+require 'json'
+require 'open-uri'
+
+DOMAIN = 'https://secure.login.gov'
+
+def copy_translations(origin_uris, destination)
+  origin_uris.each do |uri|
+    filename = uri.gsub(DOMAIN, '').gsub(%r{^/}, '')
+    dest_filename = File.join(destination, filename)
+    next if File.exist?(dest_filename)
```

Comment:
We shouldn't just check if the file exists, we should also check if it's _newer_. That may require a `HEAD` request to IDP for the [`Last-Modified`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified) or [`Cache-Control`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control) headers.

Reply:
Might also want to add a check on the size to ensure the file was entirely and correctly written as well.

Reply:
It will matter if the file location has already been populated (e.g. from cache) because this script would skip downloading changes to the files. i.e. the translations could become out-of-date.

Reply:
> Doesnt seem like that header is provided by CloudFront :(

After looking over some documentation, seems the best we can probably do for an integrity check is emit a warning if the `Etag` header ([S3](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html)/[CloudFront](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html)) isn't present or doesn't match the file's MD5 hash.

Reply:
Also as an alternative to checking for the time, looks like we could also try sending the [`If-Modified-Since`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since) header.

Reply:
I think for this script, for now, we don't need to worry about optimizing the recency of the files, since like a previous comment said the overall size is small. Also each time this is run in CI it'll be a fresh build and by default federalist only does a shallow git checkout, so there's not even real file age. I would not go down this path just yet

Reply:
It looks like the file names include a portion that typically changes when a build is run, so checking for modifications may be less important that checking integrity here.

Reply:
Resolving though since we're always downloading the file now.